File: backend/internal/post/domain/post.go
package domain

import (
	"context"
	"errors"
	"time"
)

var ErrPostNotFound = errors.New("post not found")

// Post represents a blog post.
type Post struct {
	ID           uint64
	Title        string
	Content      string
	Summary      string
	ContentType  string
	CoverImage   string
	AuthorID     uint64
	Status       string
	ViewCount    int
	LikeCount    int
	CommentCount int
	CollectCount int
	Tags         []string
	PublishedAt  time.Time
	CreatedAt    time.Time
	UpdatedAt    time.Time
}

const (
	ContentTypeMarkdown = "markdown"
	ContentTypeRichText = "rich_text"

	StatusDraft     = "draft"
	StatusPending   = "pending"
	StatusPublished = "published"
	StatusRejected  = "rejected"
)

// PostRepository defines the data access interface for Post.
type PostRepository interface {
	Create(ctx context.Context, p *Post) error
	GetByID(ctx context.Context, id uint64) (*Post, error)
	Update(ctx context.Context, p *Post) error
	Delete(ctx context.Context, id uint64) error
	List(ctx context.Context, filter ListFilter) ([]*Post, int64, error)
}

// ListFilter defines filtering options for post listing.
type ListFilter struct {
	Status   string
	AuthorID uint64
	Sort     string // "new", "hot"
	Page     int
	PageSize int
}

File: frontend/types/post.ts
export interface Post {
  id: number;
  title: string;
  content?: string;
  summary: string;
  contentType: "markdown" | "rich_text";
  coverImage: string;
  authorId: number;
  authorName?: string;
  status: "draft" | "pending" | "published" | "rejected";
  viewCount: number;
  likeCount: number;
  commentCount: number;
  collectCount: number;
  tags: string[];
  publishedAt?: number;
  createdAt: number;
  updatedAt: number;
}

export interface Pagination {
  total: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

export interface PostsResponse {
  list: Post[];
  pagination: Pagination;
}

export interface CreatePostPayload {
  title: string;
  content: string;
  contentType?: "markdown" | "rich_text";
  coverImage?: string;
  tags?: string[];
}
